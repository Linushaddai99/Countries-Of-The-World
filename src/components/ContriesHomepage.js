import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import getCountries from '../redux/api';
import CountriesList from './ContriesList';

const ContriesHomepage = () => {
  const dispatch = useDispatch();
  const countries = useSelector((state) => state.countries.countries);

  const [searchItem, setSearchItem] = useState('');

  console.log(countries);

  useEffect(() => {
    if (countries.length === 0) {
      dispatch(getCountries());
    }
  }, [dispatch, countries.length]);

  const newCountries = countries.filter((country) => (
    country.name.toLowerCase().includes(searchItem.toLowerCase())
    || country.region.toLowerCase().includes(searchItem.toLowerCase())));

  const handleSearch = (e) => {
    e.preventDefault();
    setSearchItem(e.target.value);
  };

  return (
    <div>
      <h1>Where in the World</h1>
      <input type="text" name="searchItem" placeholder="search for a country" value={searchItem} onChange={handleSearch} />
      {searchItem.length ? <CountriesList countries={newCountries} />
        : <CountriesList countries={countries} />}

    </div>
  );
};

export default ContriesHomepage;
