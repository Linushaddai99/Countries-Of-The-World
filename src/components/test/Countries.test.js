import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../redux/configStore';
import getCountries from '../../redux/api';

import ContriesHomepage from '../ContriesHomepage';

describe('displays all the countries from the api', () => {
  it('should display all the countries', () => {
    const { container } = render(
      <Provider store={store}>
        <ContriesHomepage />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});

describe('Countries reducers', () => {
  jest.setTimeout(30000);
  it('should fetch items', async () => {
    await store.dispatch(getCountries());
    expect(store.getState().countries.countries.length).toEqual(250);
  });
});
