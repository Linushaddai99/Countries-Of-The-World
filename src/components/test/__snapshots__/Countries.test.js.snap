// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays all the countries from the api should display all the countries 1`] = `
<div>
  <div>
    <input
      class="search-bar"
      name="searchItem"
      placeholder="search for a country"
      type="text"
      value=""
    />
    <div
      class="countries-grid"
    />
  </div>
</div>
`;
